front/src/modules/trainschedule/components/ManageTrainSchedule/helpers/formatTimetableItemPayload.ts

```diff
@@ -34,18 +40,66 @@ export function formatTimetableItemPayload(
   };
 }
 
+// Format a PacedTrainWithDetails to a PacedTrain payload by keeping only the
+// necessary properties and formatting the date fields to ISO strings.
+function formatPacedTrainWithDetailsToPacedTrainPayload(
+  pacedTrainWithDetails: PacedTrainWithDetails
+): PacedTrain {
+  return {
+    category: pacedTrainWithDetails.category,
+    comfort: pacedTrainWithDetails.comfort,
+    constraint_distribution: pacedTrainWithDetails.constraint_distribution,
+    exceptions: pacedTrainWithDetails.exceptions,
+    initial_speed: pacedTrainWithDetails.initial_speed,
+    labels: pacedTrainWithDetails.labels,
+    margins: pacedTrainWithDetails.margins,
+    options: pacedTrainWithDetails.options,
+    paced: {
+      time_window: pacedTrainWithDetails.paced.timeWindow.toISOString(),
+      interval: pacedTrainWithDetails.paced.interval.toISOString(),
+    },
+    path: pacedTrainWithDetails.path,
+    power_restrictions: pacedTrainWithDetails.power_restrictions,
+    rolling_stock_name: pacedTrainWithDetails.rollingStock!.name,
+    schedule: pacedTrainWithDetails.schedule,
+    speed_limit_tag: pacedTrainWithDetails.speed_limit_tag,
+    start_time: pacedTrainWithDetails.startTime.toISOString(),
+    train_name: pacedTrainWithDetails.name,
+  };
+}
+
 export function formatPacedTrainPayload(
   osrdconf: OperationalStudiesConfState,
   // TODO TS2 : remove this when rollingStockName will replace rollingStockId in the store
-  rollingStockName: string
+  rollingStockName: string,
+  timetableItemToEditData?: TimetableItemToEditData
 ): PacedTrain {
   const baseTrain = formatTimetableItemPayload(osrdconf, rollingStockName);
-  return {
+
+  let pacedTrainToUpdate: PacedTrain = {
     ...baseTrain,
     paced: {
       time_window: osrdconf.timeWindow.toISOString(),
       interval: osrdconf.interval.toISOString(),
     },
-    exceptions: [],
+    exceptions: timetableItemToEditData?.originalPacedTrain?.exceptions ?? [],
   };
+
+  if (timetableItemToEditData?.originalPacedTrain && timetableItemToEditData?.occurrenceId) {
+    const originalPacedTrain = formatPacedTrainWithDetailsToPacedTrainPayload(
+      timetableItemToEditData.originalPacedTrain
+    );
+    const updatedExceptions = updatePacedTrainException(
```

Comment:
This function is quite heavy and mix 2 different steps. Do you think it's possible to split it into 2 functions ? One that will generate the new exception and one that will update the exceptions array ?
Something like this should work:
```
const exception = generateException(originalPacedTrain, newPacedTrain, occurrenceId);

const updatedExceptions = updatePacedTrainExceptions(
  originalPacedTrain.exceptions,
  exception,
  occurrenceId
);

return {
  ...originalPacedTrain,
  exceptions: updatedExceptions,
};
```